packages/shared/src/components/post/PostActions.tsx

```diff
@@ -42,95 +37,27 @@ export function PostActions({
   onBookmark,
   origin = Origin.ArticlePage,
 }: PostActionsProps): ReactElement {
-  const { trackEvent } = useContext(AnalyticsContext);
-  const { user, showLogin } = useContext(AuthContext);
-  const { updatePost } = useUpdatePost();
   const { data, onShowPanel, onClose } = useBlockPostPanel(post);
   const { showTagsPanel } = data;
-  const onUpvotePostMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.upvote(post),
-  });
-  const onDownvoteMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.downvote(post),
-  });
-  const onCancelDownvoteMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.cancelDownvote(post),
-  });
-  const { upvotePost, cancelPostUpvote, downvotePost, cancelPostDownvote } =
-    useVotePost({
-      onUpvotePostMutate: (params) => {
-        onClose(true);
-        return onUpvotePostMutate(params);
-      },
-      onCancelPostUpvoteMutate: updatePost({
-        id: post.id,
-        update: mutationHandlers.cancelUpvote(post),
-      }),
-      onDownvotePostMutate: (params) => {
-        onShowPanel();
-        return onDownvoteMutate(params);
-      },
-      onCancelPostDownvoteMutate: (params) => {
-        onClose(true);
-        return onCancelDownvoteMutate(params);
-      },
-    });
 
-  const toggleUpvote = () => {
-    if (user) {
-      if (post?.userState?.vote === UserPostVote.Up) {
-        trackEvent(
-          postAnalyticsEvent(AnalyticsEvent.RemovePostUpvote, post, {
-            extra: { origin },
-          }),
-        );
-        return cancelPostUpvote({ id: post.id });
-      }
-      if (post) {
-        trackEvent(
-          postAnalyticsEvent(AnalyticsEvent.UpvotePost, post, {
-            extra: { origin },
-          }),
-        );
-        return upvotePost({ id: post.id });
-      }
-    } else {
-      showLogin(AuthTriggers.Upvote);
-    }
-    return undefined;
-  };
+  const { toggleUpvote, toggleDownvote } = useVotePost();
 
-  const toggleDownvote = () => {
-    if (!post) {
-      return;
+  const onToggleUpvote = async () => {
+    if (post?.userState?.vote === UserPostVote.None) {
+      onClose(true);
     }
 
-    if (!user) {
-      showLogin(AuthTriggers.Downvote);
-
-      return;
-    }
-
-    if (post?.userState?.vote === UserPostVote.Down) {
-      trackEvent(
-        postAnalyticsEvent(AnalyticsEvent.RemovePostDownvote, post, {
-          extra: { origin },
-        }),
-      );
+    await toggleUpvote({ post, origin });
+  };
```

Comment:
Not something to address here, but a more general question on the usage of `useCallback` and `useMemo`. I'm noticing these are usually not used In the project, is this intentional or not considered necessary?

Reply:
We have some guidelines for usage of those in our hooks https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/248020993/Guidelines#Creating-custom-hooks, this is because hooks are used in multiple places as parts of core logic and we want them to be defined and as optimized as possible. 

While above guidelines can be applied to all components we mostly go by "optimize when needed" to avoid any premature optimizations.